core/lib/spree/core/validators/email.rb

```diff
@@ -1,11 +1,99 @@
+require 'resolv'
+require 'mail'
+
 class EmailValidator < ActiveModel::EachValidator
+  LOCAL_PART_SPECIAL_CHARS = /[\!\#\$\%\&\'\*\-\/\=\?\+\-\^\_\`\{\|\}\~]/
+  # supports all possible domains https://iwantmyname.com/domains/domain-name-registration-list-of-extensions
+  # reference from https://en.wikipedia.org/wiki/Email_address
   def validate_each(record, attribute, value)
-    unless value =~ %r{\A(([A-Za-z0-9]+_+)|
-                          ([A-Za-z0-9]+\-+)|
-                          ([A-Za-z0-9]+\.+)|
-                          ([A-Za-z0-9]+\++))*[A-Za-z0-9_]+@((\w+\-+)|
-                          (\w+\.))*\w{1,63}\.[a-zA-Z]{2,15}\z}xi
-      record.errors.add(attribute, :invalid, { value: value }.merge!(options))
+    begin
+      m = Mail::Address.new(value)
+      r = m.domain && m.address == value
+      r &&= validate_local(m.local)
+      r &&= validate_domain(m.domain)
+      r &&= validate_mx(m.domain) if options[:check_mx]
+    rescue StandardError
+      r = false
     end
+    record.errors.add(attribute, :invalid, { value: value }.merge!(options)) unless r
```

Comment:
In my opinion, the email error should be added to ```record.errors``` as soon as any of below lines return false:

```
r = m.domain && m.address == value
r &&= validate_local(m.local)
r &&= validate_domain(m.domain)
r &&= validate_mx(m.domain) if options[:check_mx]
```
Since the ```def validate_each``` performs logical conjunction here, every single line execution returning false results in validatiion failure so we can skip further checks.